src/main/java/org/emela/chautari/service/impl/RentalServiceImpl.java

```diff
@@ -0,0 +1,157 @@
+        //this.addressService = addressService;
+        this.userService = userService;
+    }
+
+    @Override
+    public RentalItemResponse createRental(RentalItemRequest rentalItemRequest) {
+        RentalEntity rentalEntity;
+        try {
+            log.info("Rental creation request has received for user-id {} ", rentalItemRequest.getUserId());
+            rentalEntity = rentalEntityMapper.mapToRentalEntity(rentalItemRequest);
+            rentalEntity = rentalEntityRepository.save(rentalEntity);
+            mapAndSaveRentalAssociatedEntity(rentalItemRequest, rentalEntity);
+        } catch (Exception e) {
+            log.error("Rental creation failed", e);
+            throw new RentalServiceException(e);
+        }
+        log.debug("Rental Item has created successfully rental-id {} for user-id {}",
+                rentalEntity.getRentalId(), rentalItemRequest.getUserId());
+        return new RentalItemResponse().rentalId(rentalEntity.getRentalId().toString());
+    }
+
+    private void mapAndSaveRentalAssociatedEntity(RentalItemRequest rentalItemRequest, RentalEntity rentalEntity) {
+        //TODO: Uncomment when userService ready
+        //UserEntity userEntity = userService.getUserEntity(rentalItemRequest.getUserId());
+        //UserEntity userEntity = new UserEntity();
+        rentalAvailabilityService.createRentalItemAvailabilityEntity(rentalItemRequest.getAvailability(),
+                rentalEntity);
+        rentalFeatureService.createRentalItemFeatureEntity(rentalItemRequest.getFeatures(), rentalEntity);
+        rentalPreferenceService.createRentalItemPreferenceEntity(rentalItemRequest.getPreferences(), rentalEntity);
+        // addressService.saveAddress(rentalItemRequest.getLocation(), userEntity);
+    }
+
+    @Override
+    public RentalItemResponse deleteRental(String rentalId) {
+        log.info("Rental delete request has received for rental-id {} ", rentalId);
+        try {
+            RentalEntity rentalEntity = rentalEntityRepository.findByRentalId(UUID.fromString(rentalId))
+                    .orElseThrow(ResourceNotFoundException::new);
+            rentalEntityRepository.delete(rentalEntity);
+        } catch (Exception e) {
+            log.error("Delete rental Item failed", e);
+            throw new RentalServiceException(e);
+        }
+        return new RentalItemResponse().rentalId(rentalId);
+    }
+
+    @Override
+    public RentalItemDetail getRentalItemDetail(String rentalId) {
+        log.info("Rental get request has received for rental-id {} ", rentalId);
+        RentalItemDetail rentalItemDetail;
+        try {
+            RentalEntity rentalEntity = rentalEntityRepository.findByRentalId(UUID.fromString(rentalId))
+                    .orElseThrow(ResourceNotFoundException::new);
+            rentalItemDetail = rentalEntityMapper.mapToRentalItemDetail(rentalEntity);
+            rentalItemDetail.setAvailability(rentalAvailabilityService.
```

Comment:
why are we setting these 3 properties, rentalItemDetail should have all the associated data?